src/mlpack/methods/ann/layer/alpha_dropout.hpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * @file alpha_dropout.hpp
+ * @author Dakshit Agrawal
+ *
+ * Definition of the Alpha-Dropout class, which implements a regularizer that
+ * randomly sets units to alpha-dash to prevent them from co-adapting and
+ * makes an affine transformation so as to keep the mean and variance of
+ * outputs at their original values.
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LAYER_ALPHA_DROPOUT_HPP
+#define MLPACK_METHODS_ANN_LAYER_ALPHA_DROPOUT_HPP
+
+#include <mlpack/prereqs.hpp>
+
+// To use power function.
+#include <cmath>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+* The alpha - dropout layer is a regularizer that randomly with probability 'ratio'
+* sets input values to alpha_dash.  An affine transformation is applied to the inputs.
+* In the deterministic mode (during testing), the layer just gives the output.
+*
+* Note: During training you should set deterministic to false and during
+* testing you should set deterministic to true.
+*
+* For more information, see the following.
+*
+* @code
+* @article{Klambauer2017,
+*   author  = {Gunter Klambauer and Thomas Unterthiner and
+*              Andreas Mayr},
+*   title   = {Self-Normalizing Neural Networks},
+*   journal = {Advances in Neural Information Processing Systems},
+*   year    = {2017}
+* }
+* }
+* @endcode
+*
+* @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+*         arma::sp_mat or arma::cube).
+* @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+*         arma::sp_mat or arma::cube).
+*/
+template <
+        typename InputDataType = arma::mat,
+        typename OutputDataType = arma::mat
+>
+class Alpha_Dropout
```

Comment:
Please use camel casing for all names.

Reply:
I will keep that in mind.  Thanks for pointing it out.